js/lex.go

```diff
@@ -289,35 +288,35 @@ func (l *Lexer) consumeUnicodeEscape() bool {
 	return true
 }
 
-////////////////////////////////////////////////////////////////
-
-func (l *Lexer) consumeSingleLineToken(skip int) {
-	l.r.Move(skip)
+func (l *Lexer) consumeSingleLineComment() {
 	for {
 		c := l.r.Peek(0)
 		if c == '\r' || c == '\n' || c == 0 {
 			break
 		} else if c >= 0xC0 {
-			mark := l.r.Pos()
+			//mark := l.r.Pos()
 			if r, _ := l.r.PeekRune(0); r == '\u2028' || r == '\u2029' {
-				l.r.Rewind(mark)
+				//l.r.Rewind(mark)
 				break
 			}
 		}
 		l.r.Move(1)
```

Comment:
Hm yes, this might slightly improve performance. Usually UTF-8 codepoints are very rare (compared to ASCII), so that any execution paths for UTF-8 might be in the way for ASCII (as in CPU cache lines being missed). So if there is an improvement it would still be very small. Usually these performance predictions are wrong because it is so non-trivial ;-)

I will run a benchmark to prove it